sdk/src/androidTest/java/ly/count/android/sdk/scUP_UserProfileTests.java

```diff
@@ -0,0 +1,492 @@
+
+        countly.deviceId().changeWithoutMerge("new_device_id");
+
+        // first user property request + 3 events + user property request with light_mode
+        ModuleUserProfileTests.validateUserProfileRequest(0, 3, TestUtils.map(), TestUtils.map("theme", "dark_mode"));
+        ModuleUserProfileTests.validateUserProfileRequest(2, 3, TestUtils.map(), TestUtils.map("theme", "light_mode"));
+    }
+
+    /**
+     * 1. 200_CNR_A
+     * Init SDK
+     * sendUserProperties
+     * sendUserData
+     * Check request queue:
+     * - There can be a began session request
+     * - There should be no user property request
+     * - There should be no user data request
+     */
+    @Test
+    public void UP_200_CNR_A() {
+        Countly countly = new Countly().init(TestUtils.createBaseConfig());
+
+        sendUserProperties(countly);
+        sendUserData(countly);
+
+        TestUtils.assertRQSize(0);
+    }
+
+    /**
+     * 2. 201_CR_CG_A => same as 200
+     */
+    @Test
+    public void UP_201_CR_CG_A() {
+        Countly countly = new Countly().init(TestUtils.createBaseConfig().setRequiresConsent(true).giveAllConsents());
+
+        sendUserProperties(countly);
+        sendUserData(countly);
+
+        TestUtils.assertRQSize(1);
+        ModuleConsentTests.validateAllConsentRequest(TestUtils.commonDeviceId, 0);
+    }
+
+    /**
+     * 3. 202_CR_CNG_A => same as 200 but there should be no request except consent and location
+     */
+    @Test
+    public void UP_202_CR_CNG_A() {
+        Countly countly = new Countly().init(TestUtils.createBaseConfig().setRequiresConsent(true));
+
+        sendUserProperties(countly);
+        sendUserData(countly);
+
+        TestUtils.assertRQSize(2);
+        ModuleSessionsTests.validateSessionConsentRequest(0, false, TestUtils.commonDeviceId);
+        TestUtils.validateRequest(TestUtils.commonDeviceId, TestUtils.map("location", ""), 1);
+    }
+
+    /**
+     * 4. 203_CNR_A_events
+     * Init SDK
+     * RecordBasicEvent A
+     * RecordBasicEvent B
+     * sendSameData
+     * RecordBasicEvent C
+     * sendSameData
+     * RecordBasicEvent D
+     * sendSameData
+     * RecordBasicEvent E
+     * Check requests queue:
+     * 1. Begin session
+     * 2. Event A and B
+     * 3. User Property a12345 = 4
+     * 4. Event C
+     * 5. User Property a12345 = 4
+     * 6. Event D
+     * 7. User Property a12345 = 4
+     * Check event queue:
+     * 1. Event E
+     */
+    @Test
+    public void UP_203_CNR_A_events() throws JSONException {
+        Countly countly = new Countly().init(TestUtils.createBaseConfig());
+
+        countly.events().recordEvent("A");
+        countly.events().recordEvent("B");
+        sendSameData(countly);
+        countly.events().recordEvent("C");
+        sendSameData(countly);
+        countly.events().recordEvent("D");
+        sendSameData(countly);
+        countly.events().recordEvent("E");
+
+        TestUtils.assertRQSize(6);
```

Comment:
It is because lifecylye not started yet
![image](https://github.com/Countly/countly-sdk-android/assets/57103426/1a094511-8c14-4746-93e8-2fe06876af68)